subsys/bluetooth/audio/tbs_client.c

```diff
@@ -440,160 +574,129 @@ static void call_state_notify_handler(struct bt_conn *conn,
-					      const void *data, uint16_t length)
+static uint8_t termination_reason_notify_handler(struct bt_conn *conn,
+						 struct bt_gatt_subscribe_params *params,
+						 const void *data, uint16_t length)
 {
+	struct bt_tbs_instance *tbs_inst = CONTAINER_OF(params,
+							struct bt_tbs_instance,
+							termination_sub_params);
 	struct bt_tbs_terminate_reason reason;
 
-	LOG_DBG("");
+	if (data == NULL) {
+		unsubscribed(tbs_inst, params);
+		return BT_GATT_ITER_STOP;
+	}
+
+	LOG_HEXDUMP_DBG(data, length, "");
 
 	if (length == sizeof(reason)) {
 		(void)memcpy(&reason, data, length);
-		LOG_DBG("ID 0x%02X, reason %s", reason.call_index,
+		LOG_DBG("%p index %u ID 0x%02X, reason %s", tbs_inst,
+			tbs_index(conn, tbs_inst), reason.call_index,
 			bt_tbs_term_reason_str(reason.reason));
 
 		if (tbs_client_cbs != NULL && tbs_client_cbs->termination_reason != NULL) {
 			tbs_client_cbs->termination_reason(conn, 0, tbs_index(conn, tbs_inst),
 							   reason.call_index, reason.reason);
 		}
 	}
+
+	return BT_GATT_ITER_CONTINUE;
 }
 
 #if defined(CONFIG_BT_TBS_CLIENT_INCOMING_CALL)
-static void in_call_notify_handler(struct bt_conn *conn,
-				   const struct bt_tbs_instance *tbs_inst,
-				   const void *data, uint16_t length)
+static uint8_t in_call_notify_handler(struct bt_conn *conn,
+				      struct bt_gatt_subscribe_params *params,
+				      const void *data, uint16_t length)
 {
-	const char *uri = parse_string_value(data, length,
-					     CONFIG_BT_TBS_MAX_URI_LENGTH);
+	struct bt_tbs_instance *tbs_inst = CONTAINER_OF(params,
+							struct bt_tbs_instance,
+							incoming_call_sub_params);
+	const char *uri;
 
-	LOG_DBG("%s", uri);
+	if (data == NULL) {
+		unsubscribed(tbs_inst, params);
+		return BT_GATT_ITER_STOP;
+	}
+
+	LOG_HEXDUMP_DBG(data, length, "");
+
+	uri = parse_string_value(data, length, CONFIG_BT_TBS_MAX_URI_LENGTH);
+
+	LOG_DBG("%p index %u %s", tbs_inst, tbs_index(conn, tbs_inst), uri);
 
 	if (tbs_client_cbs != NULL && tbs_client_cbs->remote_uri != NULL) {
 		tbs_client_cbs->remote_uri(conn, 0, tbs_index(conn, tbs_inst), uri);
 	}
+
+	return BT_GATT_ITER_CONTINUE;
 }
 #endif /* defined(CONFIG_BT_TBS_CLIENT_INCOMING_CALL) */
 
 #if defined(CONFIG_BT_TBS_CLIENT_CALL_FRIENDLY_NAME)
-static void friendly_name_notify_handler(struct bt_conn *conn,
-					 const struct bt_tbs_instance *tbs_inst,
-					 const void *data, uint16_t length)
-{
-	const char *name = parse_string_value(data, length,
-					      CONFIG_BT_TBS_MAX_URI_LENGTH);
-
-	LOG_DBG("%s", name);
-
-	if (tbs_client_cbs != NULL && tbs_client_cbs->friendly_name != NULL) {
-		tbs_client_cbs->friendly_name(conn, 0, tbs_index(conn, tbs_inst), name);
-	}
-}
-#endif /* defined(CONFIG_BT_TBS_CLIENT_CALL_FRIENDLY_NAME) */
-
-/** @brief Handles notifications and indications from the server */
-static uint8_t notify_handler(struct bt_conn *conn,
-			      struct bt_gatt_subscribe_params *params,
-			      const void *data, uint16_t length)
```

Comment:
What was the reasoning for removing this? 

It seems that the lines
```c
	struct bt_tbs_instance *tbs_inst = CONTAINER_OF(params,
							struct bt_tbs_instance,
							incoming_call_sub_params);

	if (data == NULL) {
		unsubscribed(tbs_inst, params);
		return BT_GATT_ITER_STOP;
	}
```

Is now done for each characteristic, instead of just one place. Runtime wise it won't make a difference, but it does increase the code size a bit, right, as well as more effort to maintain. 

Arguably it does improve readability though.